bulk_indexer_test.go

```diff
@@ -250,6 +250,39 @@ func TestPipeline(t *testing.T) {
 	require.Equal(t, int64(2), stat.Indexed)
 }
 
+func TestQueryParams(t *testing.T) {
+	client := docappendertest.NewMockElasticsearchClient(t, func(w http.ResponseWriter, r *http.Request) {
+		queryParams := r.URL.Query()
+		// requires query param to have pipeline
+		require.True(t, queryParams.Has("pipeline"))
+		require.Equal(t, "test-pipeline", queryParams.Get("pipeline"))
+
+		_, result, _, _, _ := docappendertest.DecodeBulkRequestWithStatsAndDynamicTemplatesAndPipelines(r)
+		err := json.NewEncoder(w).Encode(result)
+		require.NoError(t, err)
+	})
+
+	indexer, err := docappender.NewBulkIndexer(docappender.BulkIndexerConfig{
+		Client: client,
+		QueryParams: map[string][]string{
+			"pipeline": []string{"test-pipeline"},
```

Comment:
since there is special handling in implementation around multiple string values under a key, let's test that as well, although I don't expect it in practice and I don't know what's ES behavior around that.